run-tests-with-docker.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+set -e
+
+# We aren't properly shutting down these services. Therefore
+# the internal data store might get corrupted. Completely clean up
+# existing images to ensure that we will not hunt ghost bugs.
+docker-compose kill
```

Comment:
I assume the idea is to execute `run-tests-with-docker.sh` every time one would execute `npm test`? Since this will terminate and remove the containers, won't this be a lot slower?

Reply:
Yes, that's the idea. I'm a bit surprised about this question because this script is just literally  following the advice that was in CONTRIBUTING.md before (written by you in 2016, I think :-) ). 

Btw, on my machine this takes ~12 seconds for spinning up the containers, while the full test suite takes ~ 2 minutes.

I can also remove this from this PR if you prefer and put it back into my .git/info/exclude to keep the script locally.

Reply:
Discussed this face to face. We will keep this change for now. Should once discover any problems, one can always just execute a part of the script.